src/type-coercions.md

```diff
@@ -15,55 +15,59 @@ sites are:
 
 * `let` statements where an explicit type is given.
 
-   For example, `42` is coerced to have type `i8` in the following:
+   For example, `&mut 42` is coerced to have type `&i8` in the following:
 
    ```rust
-   let _: i8 = 42;
+   let _: &i8 = &mut 42;
    ```
 
-* `static` and `const` statements (similar to `let` statements).
+* `static` and `const` item declarations (similar to `let` statements).
 
 * Arguments for function calls
 
   The value being coerced is the actual parameter, and it is coerced to
   the type of the formal parameter.
 
-  For example, `42` is coerced to have type `i8` in the following:
+  For example, `&mut 42` is coerced to have type `&i8` in the following:
 
   ```rust
-  fn bar(_: i8) { }
+  fn bar(_: &i8) { }
 
   fn main() {
-      bar(42);
+      bar(&mut 42);
   }
   ```
 
   For method calls, the receiver (`self` parameter) can only take advantage
   of [unsized coercions](#unsized-coercions).
 
-* Instantiations of struct or variant fields
+* Instantiations of struct, union, or enum variant fields
 
-  For example, `42` is coerced to have type `i8` in the following:
+  For example, `&mut 42` is coerced to have type `&i8` in the following:
 
   ```rust
-  struct Foo { x: i8 }
+  struct Foo<'a> { x: &'a i8 }
 
   fn main() {
-      Foo { x: 42 };
+      Foo { x: &mut 42 };
   }
   ```
-
-* Function results, either the final line of a block if it is not
+  
+* Function results&ndash;either the final line of a block if it is not
   semicolon-terminated or any expression in a `return` statement
 
-  For example, `42` is coerced to have type `i8` in the following:
+  For example, `x` is coerced to have type `&dyn Display` in the following:
 
   ```rust
-  fn foo() -> i8 {
-      42
+  use std::fmt::Display;
+  fn foo(x: &u32) -> &dyn Display {
+      x
   }
   ```
 
+* The [as] type cast operator can also explicitly perform type coersion.
```

Comment:
```suggestion
* The [as] type cast operator can also explicitly perform type coercion.
```

I also feel like `as` was originally not included here because it isn't quite the same as a "coercion site".  What is the reason for adding it?

Reply:
Didn't mean to resolve this. It's not a coercion site, but it's a connected concept with a lot of overlap, and I think it's worthy of a mention.

Reply:
It might be unclear of me to list it with the coersion sites, I can add a clause about "While not a coersion site..." or maybe move it to the introduction.

Reply:
Since this list is explicitly a list of coercion sites, I don't think it should include something that is explicitly not a coercion site. Can you move it up to the introduction?  Just say something like, `For explicit type coercions, see [type cast expressions].`.

Reply:
Done!